src/main/java/com/atlassian/bitbucket/jenkins/internal/credentials/JenkinsToBitbucketCredentials.java

```diff
@@ -23,4 +26,30 @@
      * @param credentials, credentials
      */
     BitbucketCredentials toBitbucketCredentials(Credentials credentials);
+
+    /**
+     * For every Bitbucket instance configured on Jenkins, we have
+     * 1. Job credentials for bitbucket server which is configured by user while creating/modifying new jobs.
+     * 2. Global credentials for bitbucket server which is configured by global admin
+     *
+     * It is possible to not specify Job credentials while configuring a job. For bitbucket operation, we
+     * fall back to global configuration. This class gives the way to create bitbucket credentials based on
+     * given optional job credentials and server configuration.
+     *
+     * @param credentials         credentials id
+     * @param serverConfiguration the server configuration
+     * @return bitbucket credentials
+     */
+    BitbucketCredentials toBitbucketCredentials(@Nullable String credentials,
+                                                BitbucketServerConfiguration serverConfiguration);
+
+    /**
+     * See {@link #toBitbucketCredentials(Credentials, BitbucketServerConfiguration)}
```

Comment:
This links to itself. Should it link to `#toBitbucketCredentials(String, BitbucketServerConfiguration)`?

Reply:
Fixed